src/Elements/Floor.cs

```diff
@@ -64,23 +65,35 @@ public List<Opening> Openings
         /// <summary>
         /// Create a floor.
         /// </summary>
-        /// <param name="profile">The profile of the floor.</param>
+        /// <param name="polygon">The polygon representing the floor perimeter.</param>
         /// <param name="elementType">The floor type of the floor.</param>
         /// <param name="elevation">The elevation of the top of the floor.</param>
         /// <param name="transform">The floor's transform. If set, this will override the floor's elevation.</param>
         /// <param name="openings">An array of openings in the floor.</param>
-        public Floor(Polygon profile, FloorType elementType, double elevation = 0.0, Transform transform = null, List<Opening> openings = null)
+        /// <param name="extrudeUp">Indicates the polygon extrusion direction along either the positive or negative z-axis. Default is positive.</param>
+        public Floor(Polygon polygon, FloorType elementType, double elevation = 0.0, Transform transform = null, List<Opening> openings = null, bool extrudeUp = true)
         {
-            this.Profile = new Profile(profile);
             if(openings != null)
             {
                 this._openings = openings;
             }
             this.Elevation = elevation;
             this.ElementType = elementType;
             var thickness = elementType.Thickness();
-            this.Transform = transform != null ? transform : new Transform(new Vector3(0, 0, elevation));
-            this.ExtrudeDirection = Vector3.ZAxis;
+            this.Transform = transform ?? new Transform(new Vector3(0, 0, elevation));
+            if (extrudeUp)
+            {
+                this.Profile = new Profile(polygon);
+                this.ExtrudeDirection = Vector3.ZAxis;
+            }
+            else
```

Comment:
If we don't allow the option to extrude up, this condition can just be replaced with `Polygon.Reversed()`

This whole conversation makes me wonder if we want to change the construction of floor to be plane-based. So that you provide a polygon and a plane, and it projects the polygon onto the plane and extrudes down from the plane. Then you could create many floors using the same polygon, but different planes.

Reply:
I like that idea. After I revise this, I'll open another branch.